src/openzaak/components/documenten/api/serializers.py

```diff
@@ -438,3 +442,55 @@ class Meta:
             "url": {"lookup_field": "uuid"},
             "informatieobject": {"validators": [IsImmutableValidator()]},
         }
+
+
+class ObjectInformatieObjectSerializer(serializers.HyperlinkedModelSerializer):
+    informatieobject = EnkelvoudigInformatieObjectHyperlinkedRelatedField(
+        view_name="enkelvoudiginformatieobject-detail",
+        lookup_field="uuid",
+        queryset=EnkelvoudigInformatieObject.objects,
+        help_text=get_help_text(
+            "documenten.ObjectInformatieObject", "informatieobject"
+        ),
+    )
+    object = LengthHyperlinkedRelatedField(
+        min_length=1,
+        max_length=1000,
+        view_name="",
+        queryset=QuerySet(),
+        lookup_field="uuid",
+        help_text=_(
+            "URL-referentie naar het gerelateerde OBJECT (in deze of een andere API)."
+        ),
+    )
+
+    def set_object_properties(self, object_type):
+        object_field = self.fields["object"]
+        if object_type == "besluit":
+            object_field.view_name = "besluit-detail"
+            object_field.queryset = Besluit.objects
+        else:
+            object_field.view_name = "zaak-detail"
+            object_field.queryset = Zaak.objects
+
+    def to_internal_value(self, data):
+        object_type = data.get("object_type")
```

Comment:
Will `data` always include `object_type`? I don't think so if a partial update omits it.

It's safer to do:

```suggestion
        object_type = data.get("object_type", self.instance.object_type)
```

Reply:
but OIO resource doesn't support `update` and `partitial_update` operations